### DISCUSSION TITLE
repeated submessages

### DISCUSSION THREAD

**Discussion by gregoryyoung** [protobuf-c/protobuf-c#653] [Q&A]
I don't mind going and writing the docs and PRing on this but im a bit stumped getting repeated sub messages to work. Unpack was pretty straight forward... Pack is a bit more interesting though

```
for(int i=0; i<write->num_events; i++) {
    EventStore__Client__Messages__NewEvent ev = EVENT_STORE__CLIENT__MESSAGES__NEW_EVENT__INIT;
            //snip
    msg.events[i] = &ev;
}
```

EVENT_STORE__CLIENT__MESSAGES__NEW_EVENT__INIT initializes a structure. If I try to make copies of the structure then I get an error about magic numbers.

What is the appropriate way of handling wanting to dynamically add repeated submessages?

**Comment by edmonds** [comment]
Hi, Greg:

How did you copy the structure, and what sort of error messages are you getting from the compiler?

**Comment by gregoryyoung** [comment]
just did a malloc/memcpy

no error message from compiler was a runtime error from protobuf-c about magic numbers. I can reproduce it in an hour or so when I get to the office.

**Comment by gregoryyoung** [comment]
Hmm in trying to reproduce it works fine must have been looking at it for too long
